core/src/mindustry/input/Placement.java

```diff
@@ -56,6 +56,30 @@
         return points;
     }
 
+    public static Array<Point2> followLineUntil(int startX, int startY, int endX, int endY){
+        Pools.freeAll(points);
+        points.clear();
+
+        Tile cursor = world.ltile(startX, startY);
+        Block filter = cursor.block();
+
+        while(true){
+            if(cursor.block() != filter) break;                        // keep looking until encountering a different block
+            if(points.contains(Pools.obtain(Point2.class, Point2::new).set(cursor.x, cursor.y))) break; // avoid biting its own tail
```

Comment:
```suggestion
            if(points.contains(p -> p.equals(cursor.x, cursor.y))) break; // avoid biting its own tail
```

Reply:
![bloop](https://user-images.githubusercontent.com/18253120/76897244-6131ee00-6861-11ea-86e5-c46726e8e378.png)
Fails to compile

Reply:
Addressed